src/controller/CHIPDeviceController.h

```diff
@@ -460,13 +461,34 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController, public SessionEst
 
     static void OnSessionEstablishmentTimeoutCallback(System::Layer * aLayer, void * aAppState, System::Error aError);
 
-    CHIP_ERROR SendOperationalCertificateSigningRequestCommand(NodeId remoteDeviceId);
-    CHIP_ERROR OnOperationalCertificateSigningRequest(NodeId node, const ByteSpan & csr);
-    CHIP_ERROR SendOperationalCertificate(NodeId remoteDeviceId, const ByteSpan & opCertBuf, const ByteSpan & icaCertBuf);
-    CHIP_ERROR SendTrustedRootCertificate(NodeId remoteDeviceId, const ByteSpan & certBuf);
+    CHIP_ERROR SendOperationalCertificateSigningRequestCommand(Device * device);
+    CHIP_ERROR SendOperationalCertificate(Device * device, const ByteSpan & opCertBuf, const ByteSpan & icaCertBuf);
+    CHIP_ERROR SendTrustedRootCertificate(Device * device);
+
+    CHIP_ERROR OnOperationalCredentialsProvisioningCompletion(Device * device);
+
+    static void OnCSRFailureResponse(void * context, uint8_t status);
+    static void OnOperationalCertificateSigningRequest(void * context, ByteSpan CSR, ByteSpan CSRNonce, ByteSpan VendorReserved1,
+                                                       ByteSpan VendorReserved2, ByteSpan VendorReserved3, ByteSpan Signature);
+
+    static void OnAddOpCertFailureResponse(void * context, uint8_t status);
+    static void OnOperationalCertificateAddResponse(void * context, uint8_t StatusCode, uint64_t FabricIndex, uint8_t * DebugText);
+
+    static void OnRootCertSuccessResponse(void * context);
+    static void OnRootCertFailureResponse(void * context, uint8_t status);
+
+    CHIP_ERROR ProcessOpCSR(const ByteSpan & CSR, const ByteSpan & CSRNonce, const ByteSpan & VendorReserved1,
+                            const ByteSpan & VendorReserved2, const ByteSpan & VendorReserved3, const ByteSpan & Signature);
```

Comment:
A bunch of this stuff needs to be documented, especially because it does not look anything like the spec (e.g. the VendorReserved bits in the CSR there.....)

Reply:
I thought spec had the vendor reserved bits.

<img width="896" alt="image" src="https://user-images.githubusercontent.com/61470527/118132036-19025300-b3b4-11eb-91b6-bc3bf41e150b.png">

Reply:
Added some documentation to the functions.

Reply:
Ah, I see.  I was looking at the actual OpCSR payload in the spec, which is a struct containing a struct and an octet string, not a bunch of octet strings...  So this implementation just has a very different form from the spec so far,